docs/source/user/threading-layer.rst

```diff
@@ -59,6 +59,21 @@ choosing a threading layer, the first is by selecting a threading layer that is
 safe under various forms of parallel execution, the second is through explicit
 selection via the threading layer name (e.g. ``tbb``).
 
+Setting the threading layer priority
+------------------------------------
+
+By default, The threading layers are searched in the order of
+'tbb', 'omp', and 'workqueue'.
+If you want to let Numba choice the threading layer
+based on availability automatically but with a different priority,
+it can be changed via the environment variable
+``NUMBA_THREADING_LAYER_PRIORITY`` or through assignment to
+``numba.config.THREADING_LAYER_PRIORITY``.
+
+For example, if you want Numba to choose ``omp`` first if available,
+then ``tbb`` and so on, you can set either
+``NUMBA_THREADING_LAYER_PRIORITY="omp tbb workqueue"`` or
+``numba.config.THREADING_LAYER_PRIORITY = ["omp", "tbb", "workqueue"]``.
```

Comment:
```suggestion
By default the threading layers are searched in the order of ``'tbb'``,
``'omp'``, then ``'workqueue'``. To change this search order whilst
maintaining the selection of a threading layer based on availability, the
environment variable :envvar:`NUMBA_THREADING_LAYER_PRIORITY` can be used.

For example, to instruct Numba to choose ``omp`` first if available,
then ``tbb`` and so on, set the environment variable as
``NUMBA_THREADING_LAYER_PRIORITY="omp tbb workqueue"``.
```

I suggest not mentioning assignment to the config variable as it leads to confusion in practice as it's mutating a global that could well have already been read (e.g. if a user specifies a signature to a module level `@njit(parallel=True)` function.

Reply:
I think with the noted addition of similar caveats about setting it ahead of compilation this is ok to add.